presto-hive/src/main/java/com/facebook/presto/hive/HiveClusteringBucketFunction.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.hive.clustering.HiveClustering;
+import com.facebook.presto.hive.clustering.MortonCode;
+import com.facebook.presto.spi.BucketFunction;
+
+import java.util.List;
+
+public class HiveClusteringBucketFunction
+        implements BucketFunction
+{
+    private final List<Integer> clusterCount;
+    private final List<String> clusteredBy;
+    private final List<Object> distribution;
+    private final List<Type> types;
+    private final HiveClustering clustering;
+
+    // TODO: Add logic to generate Z-Order based on these parameters.
+    public static BucketFunction createHiveClusteringBucketFunction(
+            List<Integer> clusterCount,
+            List<String> clusteredBy,
+            List<Object> distribution,
+            List<Type> types)
+    {
+        return new HiveClusteringBucketFunction(
+                clusterCount, clusteredBy, distribution, types);
+    }
+
+    private HiveClusteringBucketFunction(
+            List<Integer> clusterCount,
+            List<String> clusteredBy,
+            List<Object> distribution,
+            List<Type> types)
+    {
+        this.clusterCount = clusterCount;
+        this.clusteredBy = clusteredBy;
+        this.types = types;
+        this.distribution = distribution;
+        this.clustering = new MortonCode(clusterCount, clusteredBy, distribution, types);
+    }
+
+    // TODO: Return Z-Order Address.
+    public int getBucket(Page page, int position)
+    {
+        return clustering.getCluster(page, position);
```

Comment:
This method is supposed to calculate bucket number,   is cluster id similar to bucket id?  

What do you think of adding `getPartitionFunction` to interface `ConnectorNodePartitioningProvider`
and adding `ClusterPartitionFunction` that implements `PartitionFunction`